warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java

```diff
@@ -38,11 +37,9 @@ public void parse(Key k) {
     }
     
     public String getDatatype() {
-        if (datatype == null) {
-            if (nulls.size() > 1) {
-                int stop = nulls.get(0);
-                datatype = new String(backing, 0, stop);
-            }
+        if (datatype == null && nulls.size() > 1) {
```

Comment:
This isn't a deal breaker but right now, we could have an invalid tf key but still allow `getDatatype()` to return something which may or may not be valid.
What do you think about all of the getXXX() methods instead doing:

```
if (datatype == null && isValid()) 
  // parse out field
```
We could do the same for uid and value, etc. That way, if it's invalid, we just return null for everything.

Reply:
Simple enough -- done